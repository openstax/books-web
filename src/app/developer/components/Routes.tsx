import React from 'react';
import { routes } from '../..';

// tslint:disable-next-line:variable-name
const Routes: React.SFC = () => <div>
  <h1>REX Routes</h1>

  {routes.map((route) => <div key={route.name}>
    <h2>{route.name}</h2>
    path: {route.paths}
  </div>)}
</div>;

export default Routes;
