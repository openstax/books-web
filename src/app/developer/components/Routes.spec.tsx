import React from 'react';
import renderer from 'react-test-renderer';
import Routes from './Routes';

describe('Routes', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<Routes />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
