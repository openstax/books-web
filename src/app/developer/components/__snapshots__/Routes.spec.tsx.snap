// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes matches snapshot 1`] = `
<div>
  <h2>
    REX Routes
  </h2>
  <div>
    <h3>
      Developer Home
    </h3>
    path: 
    /
  </div>
  <div>
    <h3>
      Content
    </h3>
    path: 
    /books/:book/pages/:page
  </div>
  <div>
    <h3>
      NotFound
    </h3>
    path: 
    /(.*)
  </div>
</div>
`;
